src/wallet/feebumper.cpp

```diff
@@ -265,11 +259,8 @@ Result CommitTransaction(CWallet* wallet, const uint256& txid, CMutableTransacti
     wtxBumped.fTimeReceivedIsTxTime = true;
     wtxBumped.fFromMe = true;
     CValidationState state;
-    if (!wallet->CommitTransaction(wtxBumped, reservekey, g_connman.get(), state)) {
-        // NOTE: CommitTransaction never returns false, so this should never happen.
-        errors.push_back(strprintf("The transaction was rejected: %s", state.GetRejectReason()));
-        return Result::WALLET_ERROR;
-    }
+    // NOTE: CommitTransaction never returns false.
```

Comment:
According to a comment, this is about to change:
```cpp
// TODO: if we expect the failure to be long term or permanent, instead delete wtx from the wallet and return failure.

Reply:
ATM the comment is correct. The assertion below helps detecting if that changes.

Reply:
Even if we changed it to return false, it will do so very rarely. Likely never in tests. So whenever this changes we introduce a non-deterministic crash when calling an rpc and no one notices until it is out in the wild.